mathics/builtin/compilation.py

```diff
@@ -50,10 +49,10 @@ class Compile(Builtin):
     #> cf[0, -2]
      = 0.5
 
-    Loops and variable assignments are not yet supported
+    Loops and variable assignments are supported as python (not llvmlite)
+    functions
     >> Compile[{{a, _Integer}, {b, _Integer}}, While[b != 0, {a, b} = {b, Mod[a, b]}]; a]       (* GCD of a, b *)
-     : Expression While[b != 0, {a, b} = {b, Mod[a, b]}] ; a could not be compiled.
-     = Function[{Global`a, Global`b}, While[b != 0, {a, b} = {b, Mod[a, b]}] ; a]
+     =  CompiledFunction[{a, b}, a, -CompiledCode-]
```

Comment:
When I assign this to a variable and run that on anythng anything other than 0 as the `b` argument this hangs. I am guessing this definition isn't quite right?

Reply:
One other thing I note is that when I look compilation.py pyflakes is reporting a number of small lint fixes.

Reply:
Yes to both things: we need at some point to review the implementation of `Compile` in a closer way. However, this is a task for a couple of weekends. What motivated me to dig into this right now is that I needed this work for the `NIntegrate` implementation.